Extension/src/LanguageServer/client.ts

```diff
@@ -321,6 +321,10 @@ class DefaultClient implements Client {
                     // The event handlers must be set before this happens.
                     languageClient.sendRequest(QueryCompilerDefaultsRequest, {}).then((compilerDefaults: configs.CompilerDefaults) => {
                         this.configuration.CompilerDefaults = compilerDefaults;
+                        
+                        // Only register the real commands after the extension has finished initializing,
+                        // e.g. prevents empty c_cpp_properties.json from generation.
+                        registerCommands();
```

Comment:
I could delay only a subset of commands, but that didn't seem necessary to bother with.